cli/lib/kontena/cli/stacks/show_command.rb

```diff
@@ -13,16 +14,36 @@ class ShowCommand < Kontena::Command
     requires_current_master
     requires_current_master_token
 
+    option '--values', :flag, 'Output the variable-value pairs as YAML'
+    include Common::StackValuesToOption
+
     def execute
-      show_stack(name)
+      values? ? show_variables : show_stack
+    end
+
+    def stack
+      @stack ||= client.get("stacks/#{current_grid}/#{name}")
+    end
+
+    def show_variables
+      puts variable_yaml
     end
 
-    def fetch_stack(name)
-      client.get("stacks/#{current_grid}/#{name}")
+    def clean_variables
+      rejected_keys = Set.new(%w(GRID STACK PARENT_STACK PLATFORM))
```

Comment:
I don't think this is necessarily needed, these are already filtered out of the [`result['variables'] = variable_values(without_defaults: true, without_vault: true)`](https://github.com/kontena/kontena/blob/3435b98ee9cb2a7261af813e97415b40796d66b8/cli/lib/kontena/cli/stacks/yaml/reader.rb#L215) that gets stored on the server?

`DEBUG=api bundle exec bin/kontena stack show --values redis`
```json
{
   "variables" : {
      "version" : "3.0"
   },
}
```
```

Reply:
I had `PARENT_STACK` in there, maybe that doesn't get cleaned

Reply:
Right, it looks like the `PARENT_STACK` is still getting stored on the server for child stacks:

```json
   "variables" : {
      "peers" : null,
      "PARENT_STACK" : "deptest"
   },
```

IMO better to fix the `result['variables'] = variable_values(...)` to also filter that out, rather than having to filter it out everywhere else that the variables are used... that also includes:

* https://github.com/kontena/kontena/blob/050b80b74da613d3f640423e52b7dd0ed0489642/cli/lib/kontena/cli/stacks/install_command.rb#L49
* https://github.com/kontena/kontena/blob/050b80b74da613d3f640423e52b7dd0ed0489642/cli/lib/kontena/cli/stacks/upgrade_command.rb#L143

Reply:
Now filtered in reader.rb

Reply:
Does that mean that this `clean_variables` method is no longer needed, and can be removed?

Reply:
I think it can stay, if the stack has been installed before this release it will have those variables.

Reply:
You mean the `GRID`/`STACK` variables? I think those were already filtered out before sending to the server in 1.3? 

https://github.com/kontena/kontena/blob/3588a33dfc992a0fb29603002ac33bfbe047c673/cli/lib/kontena/cli/stacks/yaml/reader.rb#L146-L148